### PR TITLE
build: add latest versions of go to test matrix

### PR DESCRIPTION
**Pull request by stevvooe**
Signed-off-by: Stephen Day <stephen.day@getcruise.com>

### CODE REVIEW

**Review by AkihiroSuda — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -4,18 +4,18 @@ jobs:
...
-        go-version: ${{ matrix.go-version }}
-    - name: Checkout code
-      uses: actions/checkout@v2
-    - name: Test
-      run: go test -v ./...
-    - name: Test Blake3
-      run: go test -v ./...
-      working-directory: blake3
+      - name: Install Go
+        uses: actions/setup-go@v1
```
Comment (thaJeztah) [inline .github/workflows/test.yml:12]: While at it, can you change this to use `v2`? https://github.com/actions/setup-go#v2

Also would be good to have a `run: go version` somewhere below. I've had issues in other repositories where some environment variables weren't set correctly, and all steps were running on the same go version.

Comment (stevvooe) [reply .github/workflows/test.yml]: Upgraded it, but it looks like `go version` is run in the preamble. I'll remove that step.

File: go.mod [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module github.com/opencontainers/go-digest
 
-go 1.13
+go 1.17
```
Comment (thaJeztah) [inline go.mod:3]: Do we want to update this version to 1.17? I discovered Yesterday that they somewhat "silently" started to enforce Go versions based on this in go1.16.5.
So far, it _seems_ limited to what version can be used to manage the module, but not sure if they will enforce it in other ways;

on go1.16.4:

```bash
go mod tidy
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/golang/protobuf v1.5.2
go: downloading github.com/opencontainers/go-digest v1.0.0
```

on go1.16.5:

```bash
go mod tidy
go mod tidy: go.mod file indicates go 1.17, but maximum supported version is 1.16
```

It's not mentioned in the release notes, but is indeed in the milestone: https://github.com/golang/go/issues/46144

**Comment by AkihiroSuda** [comment]
GPG key seems expired

File: go.mod (hunk 2 above)
Comment (stevvooe) [reply go.mod:3]: Ok, I'll leave this at 1.13. I don't think we have anything using newer features than that.

**Comment by stevvooe** [comment]
> GPG key seems expired
Odd. It wasn't actually expired. I deleted and re-uploaded it and now it's working. Weird.

**Review by thaJeztah — APPROVED** [review]
LGTM

**Review by vbatts — APPROVED** [review]
🧁

**Pull request merged**